File: factorial.c
#include<stdio.h>
int main()
{
int counter=1;
int x;
int fact=1;

printf("Enter a number ");
scanf("%d",&x);

while(counter<=x){
	fact=fact*counter;
	counter++;	
}

  printf("\nfactorial is %d  ",fact);
}

File: corejava/src/Factorial.java
import java.util.Scanner;

public class Factorial {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		System.out.println("Enter a no");
		int n = sc.nextInt();
		int fact=1;
		// 1*2*3*4*5
		//5*4*3*2*1
		
		for (int i = 1; i <=n; i++) {
		  fact=fact*i;	
		}
		
		System.out.println("factorial of "+n+" is "+fact);
		
		System.out.printf("\nfactorial of %d is %d",n,fact);
		
        sc.close();
	}
}
